### PR TITLE
various fixes, cleanups and release preparation

### PR DESCRIPTION
**Pull request by akallabeth**
* enable verbose compiler warnings by default
* `pkg-config` fallback for some libraries (older ubuntu does not always ship cmake files)
* fix `restricted` keyword usage
* unify compiler diagnostic pragma, use a define that can be easily managed
* fix some compiler warnings
* update changelog
* prepare release

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test again please

**Review by mfleisz — APPROVED** [review]
LGTM

File: include/freerdp/server/audin.h [hunk 1]
```diff
@@ -159,7 +159,7 @@ extern "C"
 	/** \brief sets the supported audio formats for AUDIN server channel context.
 	 *
 	 *  \param context The context to set the formats for
-	 *  \param count The number of formats found in \formats. Use \b -1 to set to default formats
+	 *  \param count The number of formats found in \b formats. Use \b -1 to set to default formats
```
Comment (pnowack) [inline include/freerdp/server/audin.h:162]: If I compare this with the next line, shouldn't this be `\param`, instead of `\b`?

Comment (akallabeth) [reply include/freerdp/server/audin.h:162]: no, `\param` is for documenting a parameter (the name that follows) and `\b` is a formatting option

**Pull request merged**